src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -17,8 +22,25 @@ using namespace std;
 using namespace ov;
 using namespace ov::op;
 
-bool shared_node_optimization(const shared_ptr<Model>& model,
-                              const unordered_map<Node::type_info_t, bool (*)(const Node*, const Node*)>& rules) {
+using rules_t = unordered_map<Node::type_info_t, bool (*)(const Node*, const Node*)>;
+
+bool type_in_rules(const rules_t& rules, Node::type_info_t& type) {
```

Comment:
it's better to use unnamed namespace 
namespace { 
}

Reply:
Done